components/script/dom/pannernode.rs

```diff
@@ -60,6 +60,15 @@ impl PannerNode {
         if count > 2 {
             return Err(Error::NotSupported)
         }
+        if options.maxDistance < 0. {
+            return Err(Error::NotSupported)
+        }
+        if options.rolloffFactor < 0. {
+            return Err(Error::Range("rolloffFactor should be positive".into()))
+        }
+        if options.coneOuterGain < 0. || options.coneOuterGain > 360. {
```

Comment:
And this one `InvalidStateError`? https://webaudio.github.io/web-audio-api/#dom-pannernode-coneoutergain